appserver/jdbc/jdbc-config/src/main/java/org/glassfish/jdbc/config/validators/JdbcConnectionPoolValidator.java

```diff
@@ -41,204 +41,234 @@
- * 
+ *
+ * Following validations are done:
+ * <ul>
+ * <li>Validation of datasource/driver classnames when resource type is not null
+ * <li>Max pool size to be always higher than steady pool size
+ * <li>Check if statement wrapping is on when certain features are enabled.
+ * </lu>
+ *
  * @author Shalini M
  */
-public class JdbcConnectionPoolValidator
-    implements ConstraintValidator<JdbcConnectionPoolConstraint, ResourcePool> {
-    
+public class JdbcConnectionPoolValidator implements ConstraintValidator<JdbcConnectionPoolConstraint, ResourcePool> {
+
     protected ConnectionPoolErrorMessages poolFaults;
-    
+
+
+    @Override
     public void initialize(final JdbcConnectionPoolConstraint constraint) {
         this.poolFaults = constraint.value();
     }
 
-    public String getParsedVariable(String variableToRetrieve) {
 
-        String[] variableReference = variableToRetrieve.split("=");
+    @Override
+    public boolean isValid(final ResourcePool pool, final ConstraintValidatorContext context) {
+        if (!JdbcConnectionPool.class.isInstance(pool)) {
+            return true;
+        }
+        // original code allowed this. Why?
+        if (this.poolFaults == null) {
+            return true;
+        }
+        final JdbcConnectionPool jdbcPool = (JdbcConnectionPool) pool;
+        switch (poolFaults) {
+            case POOL_SIZE_STEADY:
+                return validateSteadyPoolSize(jdbcPool);
+            case POOL_SIZE_MAX:
+                return validateMaxPoolSize(jdbcPool);
+            case STMT_WRAPPING_DISABLED:
+                return validateWrapping(jdbcPool);
+            case TABLE_NAME_MANDATORY:
+                return validateTableConnectionValidation(jdbcPool);
+            case CUSTOM_VALIDATION_CLASS_NAME_MANDATORY:
+                return validateClassConnectionValidation(jdbcPool);
+            case RES_TYPE_MANDATORY:
+                return validateResourceType(jdbcPool);
+            default:
+                return true;
+        }
+    }
 
-        if (variableReference.length == 1) {
-            //We got a system variable as no split occured
-            return System.getProperty(variableReference[0]);
+
+    private boolean validateSteadyPoolSize(final JdbcConnectionPool jdbcPool) {
+        final Integer steadyPoolSize = getSteadyPoolSize(jdbcPool);
+        return steadyPoolSize != null && steadyPoolSize >= 0;
+    }
+
+
+    private boolean validateMaxPoolSize(final JdbcConnectionPool jdbcPool) {
+        final Integer steadyPoolSize = getSteadyPoolSize(jdbcPool);
+        final String propertyValue = jdbcPool.getMaxPoolSize();
+        final String value = propertyValue == null ? Constants.DEFAULT_MAX_POOL_SIZE : propertyValue;
+        final Integer maxPoolSize = toInteger(resolve(value));
+        if (maxPoolSize == null || maxPoolSize < 1) {
+            return false;
+        }
+        if (steadyPoolSize != null && steadyPoolSize > maxPoolSize) {
```

Comment:
if can be replaced by single return

Reply:
Yes, I could merge both ifs to one, but it would be less understandible. Now you see on the first sight what it does.